FactorioWebInterface/Data/DefaultAdminAccount.cs

```diff
@@ -0,0 +1,147 @@
+﻿using Microsoft.AspNetCore.Identity;
+using Serilog;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace FactorioWebInterface.Data
+{
+    public class DefaultAdminAccount
+    {
+        private readonly UserManager<ApplicationUser> _userManager;
+        private readonly string _userName = "Admin";
+
+        public DefaultAdminAccount(UserManager<ApplicationUser> userManager)
+        {
+            _userManager = userManager;
+        }
+
+        private enum AccountsNumbers
+        {
+            NoAccounts,
+            OnlyDefaultAccount,
+            DefaultIsOnlyRootAccount,
+            MultipleAccounts
+        }
+
+        public async Task SetupDefaultUserAsync()
+        {
+            string id = Constants.DefaultAdminAccount;
+
+            await ClearDefaultUserAsync(id);
+
+            switch (await OnlyAccount())
+            {
+                case AccountsNumbers.DefaultIsOnlyRootAccount:
+                case AccountsNumbers.OnlyDefaultAccount:
+                    Log.Information(Constants.DefaultAdminAccount + " could not be created, another account already exists");
+                    return;
+                case AccountsNumbers.MultipleAccounts:
+                    await ClearDefaultUserAsync(id, true);
+                    return;
+                case AccountsNumbers.NoAccounts:
+                    break;
+            }
+
+            await CreateDefaultUserAsync(id);
+        }
+
+        //Todo: Perform count directly on database
```

Comment:
Getting a list of all users almost twice just to count them seems very inefficient, especially if there is a lot of users. I think it would be desirable to use LINQ (or SQL) to perform a count directly on the database. However for our needs this should be good enough.